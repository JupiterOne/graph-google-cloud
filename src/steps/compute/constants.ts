// Steps
export const STEP_COMPUTE_INSTANCES = 'fetch-compute-instances';
export const STEP_COMPUTE_INSTANCE_SERVICE_ACCOUNT_RELATIONSHIPS =
  'build-compute-instance-service-account-relationships';
export const STEP_COMPUTE_DISKS = 'fetch-compute-disks';
export const STEP_COMPUTE_DISK_IMAGE_RELATIONSHIPS =
  'build-compute-disk-image-relationships';
export const STEP_COMPUTE_DISK_KMS_RELATIONSHIPS =
  'build-compute-disk-kms-relationships';
export const STEP_COMPUTE_REGION_DISKS = 'fetch-compute-region-disks';
export const STEP_COMPUTE_SNAPSHOTS = 'fetch-compute-snapshots';
export const STEP_COMPUTE_SNAPSHOT_DISK_RELATIONSHIPS =
  'build-compute-snapshot-disk-relationships';
export const STEP_COMPUTE_IMAGES = 'fetch-compute-images';
export const STEP_COMPUTE_IMAGE_IMAGE_RELATIONSHIPS =
  'build-image-image-relationships';
export const STEP_COMPUTE_IMAGE_KMS_RELATIONSHIPS =
  'build-compute-image-kms-relationships';
export const STEP_COMPUTE_NETWORKS = 'fetch-compute-networks';
export const STEP_COMPUTE_NETWORK_PEERING_RELATIONSHIPS =
  'fetch-compute-networks-peering-relationships';
export const STEP_COMPUTE_ADDRESSES = 'fetch-compute-addresses';
export const STEP_COMPUTE_GLOBAL_ADDRESSES = 'fetch-compute-global-addresses';
export const STEP_COMPUTE_FORWARDING_RULES = 'fetch-compute-forwarding-rules';
export const STEP_COMPUTE_GLOBAL_FORWARDING_RULES =
  'fetch-compute-global-forwarding-rules';
export const STEP_COMPUTE_SUBNETWORKS = 'fetch-compute-subnetworks';
export const STEP_COMPUTE_FIREWALLS = 'fetch-compute-firewalls';
export const STEP_COMPUTE_PROJECT = 'fetch-compute-project';
export const STEP_COMPUTE_HEALTH_CHECKS = 'fetch-compute-health-checks';
export const STEP_COMPUTE_REGION_HEALTH_CHECKS =
  'fetch-compute-region-health-checks';
export const STEP_COMPUTE_INSTANCE_GROUPS = 'fetch-compute-instance-groups';
export const STEP_COMPUTE_REGION_INSTANCE_GROUPS =
  'fetch-compute-region-instance-groups';
export const STEP_COMPUTE_LOADBALANCERS = 'fetch-compute-loadbalancers';
export const STEP_COMPUTE_REGION_LOADBALANCERS =
  'fetch-compute-region-loadbalancers';
export const STEP_COMPUTE_BACKEND_SERVICES = 'fetch-compute-backend-services';
export const STEP_COMPUTE_REGION_BACKEND_SERVICES =
  'fetch-compute-region_backend-services';
export const STEP_COMPUTE_BACKEND_BUCKETS = 'fetch-compute-backend-buckets';
export const STEP_CREATE_COMPUTE_BACKEND_BUCKET_BUCKET_RELATIONSHIPS =
  'build-compute-backend-bucket-bucket-relationships';
export const STEP_COMPUTE_TARGET_SSL_PROXIES =
  'fetch-compute-target-ssl-proxies';
export const STEP_COMPUTE_TARGET_HTTPS_PROXIES =
  'fetch-compute-target-https-proxies';
export const STEP_COMPUTE_REGION_TARGET_HTTPS_PROXIES =
  'fetch-compute-region-target-https-proxies';
export const STEP_COMPUTE_TARGET_HTTP_PROXIES =
  'fetch-compute-target-http-proxies';
export const STEP_COMPUTE_REGION_TARGET_HTTP_PROXIES =
  'fetch-compute-region-target-http-proxies';
export const STEP_COMPUTE_SSL_POLICIES = 'fetch-compute-ssl-policies';

// Entities
export const ENTITY_CLASS_COMPUTE_INSTANCE = ['Host'];
export const ENTITY_TYPE_COMPUTE_INSTANCE = 'google_compute_instance';

export const ENTITY_CLASS_COMPUTE_DISK = ['DataStore', 'Disk'];
export const ENTITY_TYPE_COMPUTE_DISK = 'google_compute_disk';

export const ENTITY_CLASS_COMPUTE_SNAPSHOT = ['Image'];
export const ENTITY_TYPE_COMPUTE_SNAPSHOT = 'google_compute_snapshot';

export const ENTITY_CLASS_COMPUTE_IMAGE = 'Image';
export const ENTITY_TYPE_COMPUTE_IMAGE = 'google_compute_image';

export const ENTITY_TYPE_COMPUTE_NETWORK = 'google_compute_network';
export const ENTITY_CLASS_COMPUTE_NETWORK = 'Network';

export const ENTITY_TYPE_COMPUTE_ADDRESS = 'google_compute_address';
export const ENTITY_CLASS_COMPUTE_ADDRESS = ['IpAddress'];

export const ENTITY_TYPE_COMPUTE_GLOBAL_ADDRESS =
  'google_compute_global_address';
export const ENTITY_CLASS_COMPUTE_GLOBAL_ADDRESS = 'IpAddress';

export const ENTITY_TYPE_COMPUTE_FORWARDING_RULE =
  'google_compute_forwarding_rule';
export const ENTITY_CLASS_COMPUTE_FORWARDING_RULE = 'Configuration';

export const ENTITY_TYPE_COMPUTE_GLOBAL_FORWARDING_RULE =
  'google_compute_global_forwarding_rule';
export const ENTITY_CLASS_COMPUTE_GLOBAL_FORWARDING_RULE = 'Configuration';

export const ENTITY_TYPE_COMPUTE_SUBNETWORK = 'google_compute_subnetwork';
export const ENTITY_CLASS_COMPUTE_SUBNETWORK = 'Network';

export const ENTITY_TYPE_COMPUTE_FIREWALL = 'google_compute_firewall';
export const ENTITY_CLASS_COMPUTE_FIREWALL = 'Firewall';

export const ENTITY_TYPE_COMPUTE_HEALTH_CHECK = 'google_compute_health_check';
export const ENTITY_CLASS_COMPUTE_HEALTH_CHECK = 'Service';

export const ENTITY_TYPE_COMPUTE_PROJECT = 'google_compute_project';
export const ENTITY_CLASS_COMPUTE_PROJECT = ['Project'];

export const ENTITY_TYPE_COMPUTE_INSTANCE_GROUP =
  'google_compute_instance_group';
export const ENTITY_CLASS_COMPUTE_INSTANCE_GROUP = 'Group';

export const ENTITY_TYPE_COMPUTE_INSTANCE_GROUP_NAMED_PORT =
  'google_compute_instance_group_named_port';
export const ENTITY_CLASS_COMPUTE_INSTANCE_GROUP_NAMED_PORT = 'Configuration';

export const ENTITY_TYPE_COMPUTE_LOAD_BALANCER = 'google_compute_url_map';
export const ENTITY_CLASS_COMPUTE_LOAD_BALANCER = 'Gateway';

export const ENTITY_TYPE_COMPUTE_BACKEND_SERVICE =
  'google_compute_backend_service';
export const ENTITY_CLASS_COMPUTE_BACKEND_SERVICE = 'Service';

export const ENTITY_TYPE_COMPUTE_BACKEND_BUCKET =
  'google_compute_backend_bucket';
export const ENTITY_CLASS_COMPUTE_BACKEND_BUCKET = 'Gateway';

export const ENTITY_TYPE_COMPUTE_TARGET_SSL_PROXY =
  'google_compute_target_ssl_proxy';
export const ENTITY_CLASS_COMPUTE_TARGET_SSL_PROXY = 'Gateway';

export const ENTITY_TYPE_COMPUTE_TARGET_HTTP_PROXY =
  'google_compute_target_http_proxy';
export const ENTITY_CLASS_COMPUTE_TARGET_HTTP_PROXY = 'Gateway';

export const ENTITY_TYPE_COMPUTE_TARGET_HTTPS_PROXY =
  'google_compute_target_https_proxy';
export const ENTITY_CLASS_COMPUTE_TARGET_HTTPS_PROXY = 'Gateway';

export const ENTITY_TYPE_COMPUTE_SSL_POLICY = 'google_compute_ssl_policy';
export const ENTITY_CLASS_COMPUTE_SSL_POLICY = 'Policy';

// Relationships
export const RELATIONSHIP_TYPE_GOOGLE_COMPUTE_INSTANCE_TRUSTS_IAM_SERVICE_ACCOUNT =
  'google_compute_instance_trusts_iam_service_account';
export const RELATIONSHIP_TYPE_GOOGLE_COMPUTE_INSTANCE_USES_DISK =
  'google_compute_instance_uses_disk';
export const RELATIONSHIP_TYPE_GOOGLE_COMPUTE_NETWORK_CONTAINS_GOOGLE_COMPUTE_SUBNETWORK =
  'google_compute_network_contains_subnetwork';
export const RELATIONSHIP_TYPE_COMPUTE_NETWORK_CONNECTS_NETWORK =
  'google_compute_network_connects_network';
export const RELATIONSHIP_TYPE_SUBNET_HAS_COMPUTE_INSTANCE =
  'google_compute_subnetwork_has_instance';
export const RELATIONSHIP_TYPE_COMPUTE_SUBNETWORK_HAS_ADDRESS =
  'google_compute_subnetwork_has_address';
export const RELATIONSHIP_TYPE_COMPUTE_SUBNETWORK_HAS_GLOBAL_ADDRESS =
  'google_compute_subnetwork_has_global_address';
export const RELATIONSHIP_TYPE_COMPUTE_NETWORK_HAS_ADDRESS =
  'google_compute_network_has_address';
export const RELATIONSHIP_TYPE_COMPUTE_NETWORK_HAS_GLOBAL_ADDRESS =
  'google_compute_network_has_global_address';
export const RELATIONSHIP_TYPE_COMPUTE_INSTANCE_USES_ADDRESS =
  'google_compute_instance_uses_address';
export const RELATIONSHIP_TYPE_FIREWALL_PROTECTS_NETWORK =
  'google_compute_firewall_protects_network';
export const RELATIONSHIP_TYPE_DISK_USES_IMAGE =
  'google_compute_disk_uses_image';
export const RELATIONSHIP_TYPE_COMPUTE_DISK_USES_KMS_CRYPTO_KEY =
  'google_compute_disk_uses_kms_crypto_key';
export const RELATIONSHIP_TYPE_DISK_CREATED_SNAPSHOT =
  'google_compute_disk_created_snapshot';
export const RELATIONSHIP_TYPE_IMAGE_CREATED_IMAGE =
  'google_compute_image_created_image';
export const RELATIONSHIP_TYPE_SNAPSHOT_CREATED_IMAGE =
  'google_compute_snapshot_created_image';
export const RELATIONSHIP_TYPE_IMAGE_USES_KMS_KEY =
  'google_compute_image_uses_kms_crypto_key';
export const RELATIONSHIP_TYPE_NETWORK_HAS_FIREWALL =
  'google_compute_network_has_firewall';
export const RELATIONSHIP_TYPE_PROJECT_HAS_INSTANCE =
  'google_compute_project_has_instance';
export const RELATIONSHIP_TYPE_INSTANCE_GROUP_HAS_COMPUTE_INSTANCE =
  'google_compute_instance_group_has_instance';
export const RELATIONSHIP_TYPE_INSTANCE_GROUP_HAS_NAMED_PORT =
  'google_compute_instance_group_has_named_port';
export const RELATIONSHIP_TYPE_BACKEND_SERVICE_HAS_INSTANCE_GROUP =
  'google_compute_backend_service_has_instance_group';
export const RELATIONSHIP_TYPE_BACKEND_SERVICE_HAS_HEALTH_CHECK =
  'google_compute_backend_service_has_health_check';
export const RELATIONSHIP_TYPE_LOAD_BALANCER_HAS_BACKEND_SERVICE =
  'google_compute_url_map_has_backend_service';
export const RELATIONSHIP_TYPE_LOAD_BALANCER_HAS_BACKEND_BUCKET =
  'google_compute_url_map_has_backend_bucket';
export const RELATIONSHIP_TYPE_BACKEND_BUCKET_HAS_STORAGE_BUCKET =
  'google_compute_backend_bucket_has_storage_bucket';

export const RELATIONSHIP_TYPE_COMPUTE_FORWARDING_RULE_CONNECTS_BACKEND_SERVICE =
  'google_compute_forwarding_rule_connects_backend_service';
export const RELATIONSHIP_TYPE_COMPUTE_FORWARDING_RULE_CONNECTS_SUBNETWORK =
  'google_compute_forwarding_rule_connects_subnetwork';
export const RELATIONSHIP_TYPE_COMPUTE_FORWARDING_RULE_CONNECTS_NETWORK =
  'google_compute_forwarding_rule_connects_network';
export const RELATIONSHIP_TYPE_COMPUTE_FORWARDING_RULE_CONNECTS_TARGET_HTTP_PROXY =
  'google_compute_forwarding_rule_connects_target_http_proxy';
export const RELATIONSHIP_TYPE_COMPUTE_FORWARDING_RULE_CONNECTS_TARGET_HTTPS_PROXY =
  'google_compute_forwarding_rule_connects_target_https_proxy';

export const RELATIONSHIP_TYPE_COMPUTE_GLOBAL_FORWARDING_RULE_CONNECTS_BACKEND_SERVICE =
  'google_compute_global_forwarding_rule_connects_backend_service';

export const RELATIONSHIP_TYPE_COMPUTE_GLOBAL_FORWARDING_RULE_CONNECTS_SUBNETWORK =
  'google_compute_global_forwarding_rule_connects_subnetwork';

export const RELATIONSHIP_TYPE_COMPUTE_GLOBAL_FORWARDING_RULE_CONNECTS_NETWORK =
  'google_compute_global_forwarding_rule_connects_network';

export const RELATIONSHIP_TYPE_COMPUTE_GLOBAL_FORWARDING_RULE_CONNECTS_TARGET_HTTP_PROXY =
  'google_compute_global_forwarding_rule_connects_target_http_proxy';

export const RELATIONSHIP_TYPE_COMPUTE_GLOBAL_FORWARDING_RULE_CONNECTS_TARGET_HTTPS_PROXY =
  'google_compute_global_forwarding_rule_connects_target_https_proxy';

export const RELATIONSHIP_TYPE_COMPUTE_FORWARDING_RULE_USES_ADDRESS =
  'google_compute_forwarding_rule_uses_address';

export const RELATIONSHIP_TYPE_LOAD_BALANCER_HAS_TARGET_HTTPS_PROXY =
  'google_compute_url_map_has_target_https_proxy';
export const RELATIONSHIP_TYPE_LOAD_BALANCER_HAS_TARGET_HTTP_PROXY =
  'google_compute_url_map_has_target_http_proxy';
export const RELATIONSHIP_TYPE_BACKEND_SERVICE_HAS_TARGET_SSL_PROXY =
  'google_compute_backend_service_has_target_ssl_proxy';
export const RELATIONSHIP_TYPE_TARGET_HTTPS_PROXY_HAS_SSL_POLICY =
  'google_compute_target_https_proxy_has_ssl_policy';
export const RELATIONSHIP_TYPE_TARGET_SSL_PROXY_HAS_SSL_POLICY =
  'google_compute_target_ssl_proxy_has_ssl_policy';

// Mapped relationships
export const MAPPED_RELATIONSHIP_FIREWALL_RULE_TYPE =
  'google_cloud_firewall_rule';
