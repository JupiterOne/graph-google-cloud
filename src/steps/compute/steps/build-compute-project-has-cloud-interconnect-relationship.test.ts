import {
  Recording,
  StepTestConfig,
  executeStepWithDependencies,
} from '@jupiterone/integration-sdk-testing';
import { STEP_COMPUTE_PROJECT_HAS_CLOUD_INTERCONNECT_RELATIONSHIP } from '../constants';
import {
  setupGoogleCloudRecording,
  getMatchRequestsBy,
} from '../../../../test/recording';
import { integrationConfig } from '../../../../test/config';
import { invocationConfig } from '../../..';

describe(`compute#${STEP_COMPUTE_PROJECT_HAS_CLOUD_INTERCONNECT_RELATIONSHIP}`, () => {
  let recording: Recording;
  afterEach(async () => {
    if (recording) await recording.stop();
  });

  test(
    STEP_COMPUTE_PROJECT_HAS_CLOUD_INTERCONNECT_RELATIONSHIP,
    async () => {
      const stepTestConfig: StepTestConfig = {
        stepId: STEP_COMPUTE_PROJECT_HAS_CLOUD_INTERCONNECT_RELATIONSHIP,
        instanceConfig: integrationConfig,
        invocationConfig: invocationConfig as any,
      };

      recording = setupGoogleCloudRecording({
        name: STEP_COMPUTE_PROJECT_HAS_CLOUD_INTERCONNECT_RELATIONSHIP,
        directory: __dirname,
        options: {
          matchRequestsBy: getMatchRequestsBy(integrationConfig),
        },
      });

      const stepResults = await executeStepWithDependencies(stepTestConfig);
      expect(stepResults.collectedRelationships.length).toBeGreaterThan(0);
    },
    500_000,
  );
});

//   describe(`compute#${STEP_COMPUTE_PROJECT_HAS_CLOUD_INTERCONNECT_RELATIONSHIP}`, () => {
//     let recording: Recording;
//     beforeEach(() => {
//       recording = setupGoogleCloudRecording({
//         directory: __dirname,
//         name: STEP_COMPUTE_PROJECT_HAS_CLOUD_INTERCONNECT_RELATIONSHIP,
//       });
//     });
//     afterEach(async () => {
//       await recording.stop();
//     });
//     function separateRelationships(collectedRelationships: Relationship[]) {
//       const { targets: directRelationships } = filterGraphObjects(
//         collectedRelationships,
//         (r) => !r._mapping,
//       ) as {
//         targets: ExplicitRelationship[];
//       };
//       return {
//         directRelationships,
//       };
//     }
//     test('should collect data', async () => {
//       const context = createMockStepExecutionContext<IntegrationConfig>({
//         instanceConfig: {
//           ...integrationConfig,
//           serviceAccountKeyFile: integrationConfig.serviceAccountKeyFile.replace(
//             'j1-gc-integration-dev-v2',
//             'j1-gc-integration-dev-v3',
//           ),
//           serviceAccountKeyConfig: {
//             ...integrationConfig.serviceAccountKeyConfig,
//             project_id: 'j1-gc-integration-dev-v3',
//           },
//         },
//       });
//       await fetchComputeProject(context);
//       await fetchCloudInterconnect(context);
//       await buildComputeProjectHasCloudInterconnectRelationship(context);
//       expect({
//         numCollectedEntities: context.jobState.collectedEntities.length,
//         numCollectedRelationships: context.jobState.collectedRelationships.length,
//         collectedEntities: context.jobState.collectedEntities,
//         collectedRelationships: context.jobState.collectedRelationships,
//         encounteredTypes: context.jobState.encounteredTypes,
//       }).toMatchSnapshot();
//       expect(
//         context.jobState.collectedEntities.filter(
//           (e) => e._type === ENTITY_TYPE_COMPUTE_PROJECT,
//         ),
//       ).toMatchGraphObjectSchema({
//         _class: ENTITY_CLASS_COMPUTE_PROJECT,
//         schema: {
//           additionalProperties: false,
//           properties: {
//             _type: { const: 'google_compute_project' },
//             _rawData: {
//               type: 'array',
//               items: { type: 'object' },
//             },
//             id: { type: 'string' },
//             name: { type: 'string' },
//             displayName: { type: 'string' },
//             isOSLoginEnabled: { type: 'boolean' },
//             kind: { type: 'string' },
//             defaultServiceAccount: { type : 'string'},
//             defaultNetworkTier: { type: 'string'},
//             createdOn: { type: 'number' },
//           },
//         },
//       });
//       expect(
//         context.jobState.collectedEntities.filter(
//           (e) => e._type === ENTITY_TYPE_CLOUD_INTERCONNECT,
//         ),
//       ).toMatchGraphObjectSchema({
//         _class: ENTITY_CLASS_CLOUD_INTERCONNECT,
//         schema: {
//           additionalProperties: false,
//           properties: {
//             _type: { const: ENTITY_TYPE_CLOUD_INTERCONNECT },
//             _rawData: {
//               type: 'array',
//               items: { type: 'object' },
//             },
//             id: { type: 'number'},
//             name: { type: 'string' },
//             location: { type: 'string' },
//             interConnectionType: { type: 'string' },
//             kind: { type: 'string' },
//             public: { type: 'boolean' },
//             adminEnabled: { type: 'boolean' },
//             createdOn: { type : 'number'}
//           },
//         },
//       });
//       const { directRelationships } = separateRelationships(
//         context.jobState.collectedRelationships,
//       );
//       expect(
//         directRelationships.filter(
//           (e) => e._type === REATIONSHIP_TYPE_COMPUTE_PROJECT_HAS_CLOUD_INTERCONNECT,
//         ),
//       ).toMatchDirectRelationshipSchema({
//         schema: {
//           properties: {
//             _class: { const: 'HAS' },
//             _type: {
//               const: REATIONSHIP_TYPE_COMPUTE_PROJECT_HAS_CLOUD_INTERCONNECT,
//             },
//           },
//         },
//       });
//     });
//   });
