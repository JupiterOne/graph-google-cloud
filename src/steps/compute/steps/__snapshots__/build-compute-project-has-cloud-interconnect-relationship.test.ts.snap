// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compute#fetch-compute-project-has-cloud-interconnect should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Project",
      ],
      "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "cloudArmorTier": "CA_STANDARD",
            "commonInstanceMetadata": {
              "fingerprint": "C_x3HYVZJUo=",
              "items": [
                {
                  "key": "enable-osconfig",
                  "value": "TRUE",
                },
                {
                  "key": "ssh-keys",
                  "value": "pgilaga:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDU0g4vACE8PrXfTGM4wUbIfQ207LxClP8p2wCI7RwGdZn+cAoMPKaHxIKck2fA1VtVWzUXehhUJbS1lrI40NQR1vH9C37shGDftEFyDORwMS2yxSWZtzUISINMzpG8Ih+iVfSAVFbjNsJpMBYDrtaLLzBOud8FDsx6scxWfI64u5hxXtjvO3rW1i4qAbYouCtMnse7xE2LUaiURpqq5ojCRlFC2bOQ+pYb0/9Jy3v6KpMnATg7Z68pWJrocnpJhkRywNwldbZLSDgff4Y5Gjbeq+Zl6qEEmyxFpqdVzB/RSM0g7eikkXZDzZHDvrXvxWiGqhBXXJ/KRfCuJG1LKzI+hl+yWvM/1qbX4qBpbnaNv+EvhS37Yu/OEqXMoUfOWqEIR//3QSf4cluhlLQ1F9GElmnicj41MFZwFq/7IQXikcrz3lfzQ5fXLmBeQ3E97z4ZSETVqHDWc04h7MdueW1OdNbKGIx7ldNAWo3TxNsV6etFTtv1QK/ZTNmqQB5fxF0= pgilaga@Pauls-MacBook-Air.local
ignacio:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDSg4sA5KWtxOoYlWtyJ+uI1f7xPQ3UMMz9tMfP3SLTQFS+ThHw4+IxhZdIy1RYug+l2YFdF5W3vOaHozE2hLarvj9hSClipoy7COn9+NkkDKLFdVajJCNxKrj1YespMBTG00frk/+t+ilz9DX0kKHklrjvQQ+TJe0R2MJa0tyAyZ4pOTXcI96YvNA0N8e/AglixvybtX1KTYj30TnrGke+vAkVa1xG0UulZ7O5At4Xd/Ue3e9SsnEO3JOUb4MLz5xKBCTY42TQp1exrccSXl4cXZV5AIuvmypkx6nrsR9K5WEgPEENCiQO+ghR6YaSnBY9Yk8w7TPI2ep0S1g2NYGjGfPCv0fhw/422utFd6amFJc1NOP8ZaT7FEpCKxVEiR30q0lARs4q1YuuQu2mx/nZ/DMQMweiXrrCfnWDRnFjvCmcT3BEtTQUa1z8MiybeVJN9MKDued6Y2dJCwBaXH0A62k4txV1b5oNAGFCPMLLRDnGzld+l+M07ub6b0CZkVM= ignacio@dbp.local",
                },
                {
                  "key": "sshKeys",
                  "value": "",
                },
              ],
              "kind": "compute#metadata",
            },
            "creationTimestamp": "2021-05-31T09:21:30.379-07:00",
            "defaultNetworkTier": "PREMIUM",
            "defaultServiceAccount": "167984947943-compute@developer.gserviceaccount.com",
            "id": "2723542235640017893",
            "kind": "compute#project",
            "name": "j1-gc-integration-dev-v3",
            "quotas": [
              {
                "limit": 10000,
                "metric": "SNAPSHOTS",
                "usage": 32,
              },
              {
                "limit": 30,
                "metric": "NETWORKS",
                "usage": 6,
              },
              {
                "limit": 500,
                "metric": "FIREWALLS",
                "usage": 22,
              },
              {
                "limit": 5000,
                "metric": "IMAGES",
                "usage": 5,
              },
              {
                "limit": 175,
                "metric": "STATIC_ADDRESSES",
                "usage": 0,
              },
              {
                "limit": 300,
                "metric": "ROUTES",
                "usage": 0,
              },
              {
                "limit": 150,
                "metric": "FORWARDING_RULES",
                "usage": 0,
              },
              {
                "limit": 500,
                "metric": "TARGET_POOLS",
                "usage": 1,
              },
              {
                "limit": 1000,
                "metric": "HEALTH_CHECKS",
                "usage": 5,
              },
              {
                "limit": 575,
                "metric": "IN_USE_ADDRESSES",
                "usage": 2,
              },
              {
                "limit": 500,
                "metric": "TARGET_INSTANCES",
                "usage": 0,
              },
              {
                "limit": 100,
                "metric": "TARGET_HTTP_PROXIES",
                "usage": 3,
              },
              {
                "limit": 100,
                "metric": "URL_MAPS",
                "usage": 3,
              },
              {
                "limit": 150,
                "metric": "BACKEND_SERVICES",
                "usage": 0,
              },
              {
                "limit": 1000,
                "metric": "INSTANCE_TEMPLATES",
                "usage": 1,
              },
              {
                "limit": 50,
                "metric": "TARGET_VPN_GATEWAYS",
                "usage": 0,
              },
              {
                "limit": 100,
                "metric": "VPN_TUNNELS",
                "usage": 1,
              },
              {
                "limit": 30,
                "metric": "BACKEND_BUCKETS",
                "usage": 1,
              },
              {
                "limit": 20,
                "metric": "ROUTERS",
                "usage": 4,
              },
              {
                "limit": 100,
                "metric": "TARGET_SSL_PROXIES",
                "usage": 0,
              },
              {
                "limit": 100,
                "metric": "TARGET_HTTPS_PROXIES",
                "usage": 2,
              },
              {
                "limit": 100,
                "metric": "SSL_CERTIFICATES",
                "usage": 2,
              },
              {
                "limit": 275,
                "metric": "SUBNETWORKS",
                "usage": 0,
              },
              {
                "limit": 100,
                "metric": "TARGET_TCP_PROXIES",
                "usage": 0,
              },
              {
                "limit": 10,
                "metric": "SECURITY_POLICIES",
                "usage": 0,
              },
              {
                "limit": 200,
                "metric": "SECURITY_POLICY_RULES",
                "usage": 0,
              },
              {
                "limit": 1000,
                "metric": "XPN_SERVICE_PROJECTS",
                "usage": 0,
              },
              {
                "limit": 150,
                "metric": "PACKET_MIRRORINGS",
                "usage": 0,
              },
              {
                "limit": 1000,
                "metric": "NETWORK_ENDPOINT_GROUPS",
                "usage": 0,
              },
              {
                "limit": 6,
                "metric": "INTERCONNECTS",
                "usage": 1,
              },
              {
                "limit": 5000,
                "metric": "GLOBAL_INTERNAL_ADDRESSES",
                "usage": 1,
              },
              {
                "limit": 50,
                "metric": "VPN_GATEWAYS",
                "usage": 2,
              },
              {
                "limit": 5000,
                "metric": "MACHINE_IMAGES",
                "usage": 0,
              },
              {
                "limit": 20,
                "metric": "SECURITY_POLICY_CEVAL_RULES",
                "usage": 0,
              },
              {
                "limit": 50,
                "metric": "EXTERNAL_VPN_GATEWAYS",
                "usage": 1,
              },
              {
                "limit": 1,
                "metric": "PUBLIC_ADVERTISED_PREFIXES",
                "usage": 0,
              },
              {
                "limit": 10,
                "metric": "PUBLIC_DELEGATED_PREFIXES",
                "usage": 0,
              },
              {
                "limit": 1024,
                "metric": "STATIC_BYOIP_ADDRESSES",
                "usage": 0,
              },
              {
                "limit": 60,
                "metric": "NETWORK_FIREWALL_POLICIES",
                "usage": 1,
              },
              {
                "limit": 150,
                "metric": "INTERNAL_TRAFFIC_DIRECTOR_FORWARDING_RULES",
                "usage": 0,
              },
              {
                "limit": 150,
                "metric": "GLOBAL_EXTERNAL_MANAGED_FORWARDING_RULES",
                "usage": 1,
              },
              {
                "limit": 150,
                "metric": "GLOBAL_INTERNAL_MANAGED_BACKEND_SERVICES",
                "usage": 0,
              },
              {
                "limit": 150,
                "metric": "GLOBAL_EXTERNAL_MANAGED_BACKEND_SERVICES",
                "usage": 0,
              },
              {
                "limit": 150,
                "metric": "GLOBAL_EXTERNAL_PROXY_LB_BACKEND_SERVICES",
                "usage": 1,
              },
              {
                "limit": 750,
                "metric": "GLOBAL_INTERNAL_TRAFFIC_DIRECTOR_BACKEND_SERVICES",
                "usage": 0,
              },
            ],
            "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3",
            "vmDnsSetting": "ZONAL_ONLY",
            "xpnProjectStatus": "UNSPECIFIED_XPN_PROJECT_STATUS",
          },
        },
      ],
      "_type": "google_compute_project",
      "createdOn": 1622478090379,
      "defaultNetworkTier": "PREMIUM",
      "defaultServiceAccount": "167984947943-compute@developer.gserviceaccount.com",
      "displayName": "j1-gc-integration-dev-v3",
      "id": "2723542235640017893",
      "isOSLoginEnabled": false,
      "kind": "compute#project",
      "name": "j1-gc-integration-dev-v3",
    },
    {
      "_class": [
        "Network",
      ],
      "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/interconnects/demo-interconnect",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "adminEnabled": true,
            "circuitInfos": [
              {
                "customerDemarcId": "",
                "googleCircuitId": "GCP-00008",
                "googleDemarcId": "VOCUS:MMR-A: Ground 7A, Frontcomms, Cab FC4, PP-HAAA12, RU21-24 FP3 Fibers 63,64 - LC",
              },
            ],
            "creationTimestamp": "2024-04-22T23:53:41.482-07:00",
            "customerName": "demo-customer",
            "description": "Example Compute Cloud Interconnect",
            "googleReferenceId": "1134185888",
            "id": "3602008363502330906",
            "interconnectType": "IT_PRIVATE",
            "kind": "compute#interconnect",
            "labelFingerprint": "42WmSpB8rSM=",
            "linkType": "LINK_TYPE_ETHERNET_10G_LR",
            "location": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/interconnectLocations/akl-zone1-1353",
            "macsecEnabled": false,
            "name": "demo-interconnect",
            "operationalStatus": "OS_UNPROVISIONED",
            "provisionedLinkCount": 1,
            "requestedLinkCount": 1,
            "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/interconnects/demo-interconnect",
            "state": "UNPROVISIONED",
          },
        },
      ],
      "_type": "google_cloud_interconnect",
      "adminEnabled": true,
      "createdOn": 1713855221482,
      "description": "Example Compute Cloud Interconnect",
      "displayName": "demo-interconnect",
      "id": "3602008363502330906",
      "interConnectType": "IT_PRIVATE",
      "kind": "compute#interconnect",
      "location": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/interconnectLocations/akl-zone1-1353",
      "name": "demo-interconnect",
      "public": true,
    },
  ],
  "collectedRelationships": [],
  "encounteredTypes": [
    "google_compute_project",
    "google_cloud_interconnect",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 0,
}
`;
